pypeit/specobjs.py

```diff
@@ -506,23 +517,35 @@ def __repr__(self):
     def __len__(self):
         return len(self.specobjs)
 
-    def build_header(self, head_fitstbl, spectrograph):
+    def build_header(self, head_fitstbl, head2d, spectrograph):
         """
         Builds the spec1d file header from the fitstbl meta data and meta data from spectrograph
+        and meta data from original header
+
         Args:
-            head_fitstbl (header):
+            head_fitstbl (:class:`astropy.io.fits.Header`):
                Header from fitstbl
-            spectrograph (object):
+            head2d (:class:`astropy.io.fits.Header`):
+               Header from the Raw file
+            spectrograph (:class:`pypeit.spectrographs.spectrograph.Spectrograph`):
                Spectrograph object
 
         Returns:
+            :class:`astropy.io.fits.Header`:
 
         """
-        header = fits.PrimaryHDU().header
+        header = initialize_header(primary=True)
```

Comment:
I don't think we need the `primary` keyword, but am happy to be proven wrong.

Reply:
In this case we do.  `initialize_header()` will not 
by default make a `PrimaryHDU().header`.

This was biting me somewhere..